### PR TITLE
Created a Dead Mans Switch Node

### PR DESCRIPTION
**Pull request by garethellis0**
Node constantly broadcasts how long it's been since it received a message on a given topic

We're going to need this to satisfy the competition requirement that once we activate the physical e-stop, the sub must remain unpowered even if the physical e-stop is closed (ie. the e-stop is reset)

### CODE REVIEW

**Comment by garethellis0** [comment]
For reference, here's a rough overview of how the system will work 
img_20180722_174803

File: src/decision/include/DeadMansSwitchNode.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+#include <std_msgs/Duration.h>
+#include <std_msgs/Empty.h>
+
+class DeadMansSwitchNode {
+  public:
+    DeadMansSwitchNode(int argc, char** argv, std::string node_name);
+
+  private:
+    /**
+     * Callback function for when a new string is received
```
Comment (reoliveira) [inline src/decision/include/DeadMansSwitchNode.h]: picky but it looks like it's not actually getting a string?

**Comment by reoliveira** [comment]
Just saw the diagram: in it the branch options are "do nothing" and "behave as normal", does that mean "do nothing" means literally "stop doing everything" or "don't react". I'm guessing the first otherwise both branches are kind of the same?

**Comment by garethellis0** [comment]
I think the rules indicate that we should cut power to the motors entirely, so literally "do nothing"

**Comment by reoliveira** [comment]
sounds good, was just clarifying

File: src/decision/include/DeadMansSwitchNode.h (hunk 1 above)
Comment (garethellis0) [reply src/decision/include/DeadMansSwitchNode.h]: Oh, good catch, that was carry over. Revamped the docstring.

**Pull request merged**